src/components/Content.js

```diff
@@ -0,0 +1,168 @@
+import { useEffect, useState } from 'react';
+import ProductsView from './ProductsView';
+import FilterView from './FilterView';
+import Cart from './Cart';
+
+
+import {
+  MDBBtn,
+  MDBModal,
+  MDBModalDialog,
+  MDBModalContent,
+  MDBModalHeader,
+  MDBModalTitle,
+  MDBModalBody,
+} from 'mdb-react-ui-kit';
+
+
+function ModalCart({showCart,setShowCart,cart,setCart}) {
+  // const [scrollableModal, setScrollableModal] = useState(true);
+  
+  return (
+      <MDBModal className="custom-modal" show={showCart} setShow={setShowCart} tabIndex='-1'>
+        <MDBModalDialog className="custom-modal-body" scrollable>
+          <MDBModalContent>
+            <MDBModalHeader>
+              <MDBModalTitle className="cart-modal-title">{"Cart"}</MDBModalTitle>
+              <MDBBtn
+                className='btn-close'
+                color='none'
+                onClick={() => {setShowCart(!showCart)}}
+              ></MDBBtn>
+            </MDBModalHeader>
+            <MDBModalBody>
+              <Cart cart={cart} setCart={setCart}/>
+            </MDBModalBody>
+          </MDBModalContent>
+        </MDBModalDialog>
+      </MDBModal>
+  );
+}
+
+function Content({showCart, setShowCart,searchString}) {
+  
+  const [products, setProducts] = useState([]);
+  const [filters, setFilters] = useState({});
+  const [cart, setCart] = useState({});
+  const [resultMessage, setResultMessage] = useState('Showing All Products');
+  const [pageNo, setPageNo] = useState(1);
+
+  const fetchProductsData = (page,filters,searchString) => {
+    var brands = filters.brand ? Array.from(filters.brand) : [];
+    var params = { brand: brands, searchString:searchString};
+    var url = '?' + ( new URLSearchParams( params ) ).toString();
+    
+    fetch("http://3.86.87.11:8080/getProducts/" + page + url)
+      .then(response => {
+        return response.json()
+      })
+      .then(data => {
+        setProducts(data);
+      })
+      .catch(err => {
+        console.log(err.message);
+      })
+
+      return () => {
+        // cleanup here
+      };
+  }
+  
+  useEffect(() => {
+    fetchProductsData(1,{},"");
+    setPageNo(1);
+  },[]);
+
+  useEffect(() => {
+    fetchProductsData(1,filters,searchString);
+    setPageNo(1);
+    
+    if(searchString !== ''){  
+      setResultMessage("Showing results for '" + searchString + "' with below filters.");
```

Comment:
@rahulmeghwal What are your thoughts with string interpolation instead of concatenation?

Reply:
String interpolations are more flexible and readable - have changed all occurrences of concatenation to interpolation.